test/correctness/simd_op_check_hvx.cpp

```diff
@@ -649,7 +651,6 @@ class SimdOpCheckHVX : public SimdOpCheckTest {
         int rfac = 4;
         RDom r(0, rfac);
         check("v*.uw += vrmpy(v*.ub,r*.ub)", hvx_width / 4, sum(u32(in_u8(rfac * x + r))));
-        check("v*.uw += vrmpy(v*.ub,r*.ub)", hvx_width / 4, sum(u32(in_u8(rfac * x + r)) * 34));
```

Comment:
Why was this removed? It looks like a special case of the one below.

Reply:
The simplifier factors the constant out of the vector reduction. The old code had some extra logic to distribute these, but this distribution is somewhat tricky to get right. If it doesn't match to a useful pattern, it could be a significant pessimization (moving a constant multiply inside a reduction loop).

At the same time, these patterns seem only marginally useful. It's basically a box filter but with a non-unit constant factor...? The case above (sum without a constant) probably covers 99% of this case.

However, I did bring back these cases in a way that checks that the reduction generates vrmpy at least for the box filter (with constant 1) part.